### ISSUE TITLE
Discussion on Reproduction Results on VizWiz and TextVqa Validation Sets

### ISSUE THREAD

**Issue by ZeroDis0bnbs** [ShawnTan86/TokenCarve#2]
Thank you for the great work. I encountered two points while reproducing Tokencarve on the validation sets:
​​1.**Unexpected accuracy trend on VizWiz validation set**​​: When reducing tokens, accuracy increased to ~66% (at 64 tokens), contrary to the expected drop from the vanilla model's ~33%. This seems counter-intuitive. Has this been observed? Any insights would be appreciated.
2.**​​Low accuracy on TextVqa validation set**​​: Accuracy was in the single digits, far below paper results. Could this be due to the evaluation metric (strict exact match?) or suboptimal default hyperparameters? Please advise on the correct evaluation protocol and effective settings.
Thanks for your time and help.

**Comment by ShawnTan86** [comment]
> Thank you for the great work. I encountered two points while reproducing Tokencarve on the validation sets:​​1.**Unexpected accuracy trend on VizWiz validation set**​​: When reducing tokens, accuracy increased to ~66% (at 64 tokens), contrary to the expected drop from the vanilla model's ~33%. This seems counter-intuitive. Has this been observed? Any insights would be appreciated. 2.**​​Low accuracy on TextVqa validation set**​​: Accuracy was in the single digits, far below paper results. Could this be due to the evaluation metric (strict exact match?) or suboptimal default hyperparameters? Please advise on the correct evaluation protocol and effective settings. Thanks for your time and help.

Hi, thanks a lot for reproducing our work and sharing these detailed observations! Really appreciate your feedback.

Regarding your points:

1. **On the VizWiz validation set:**  
   Could you please share your experimental setup? For example, are you using the **LLaVA** architecture (and which version / checkpoint)?  
   In our experiments — and also according to several recent works such as [MustDrop (arXiv:2411.10803)](https://arxiv.org/abs/2411.10803) and [FitPrune (AAAI 2025)](https://ojs.aaai.org/index.php/AAAI/article/view/34366) — the **VizWiz** dataset tends to show little sensitivity to the number of visual tokens.  
   In fact, both our 7B and 32B LLaVA-based models exhibit relatively stable performance as tokens are reduced, so an increase to 66% accuracy seems unusual.  
   We’ve also observed a similar insensitivity on the **SQA(image)** dataset, which might indicate that certain datasets are inherently less affected by token compression.

2. **On the TextVQA validation set:**  
   Our experiments consistently yielded robust results (across different machines and model sizes), so single-digit accuracy is quite unexpected.  
   Could you please confirm your **evaluation setup** — e.g., whether you’re using the same metric (exact match vs. soft matching), and which **default hyperparameters** or decoding settings you used?  
   Our released code uses the default configuration for evaluation, so reproducing the paper’s numbers should not require additional tuning.

Also, have you tried running Tokencarve on other benchmarks? 

Thanks again for taking the time to test and share your results — looking forward to your clarification so we can help pinpoint the issue!

**Comment by ZeroDis0bnbs** [comment]
Thank you again for your prompt and detailed response. We sincerely appreciate your engagement and are providing the following clarifications based on your questions.
1. Experimental Setup Clarification

-  Regarding your inquiry about our experimental setup:

​​Model Architecture & Checkpoint:​​ We used the standard ​​LLaVA-v1.5-7B​​ model, specifically the checkpoint liuhaotian/llava-v1.5-7bdownloaded directly from the official Hugging Face repository. No modifications were made to the core architecture.

- Evaluation Metric for TextVQA and VizWiz:
We confirmed that the evaluation for the TextVQA and VizWiz dataset was performed using ​​exact match. And I see now that the reason for the low accuracy in my TextVQA evaluation is likely because the associated OCR data was not used. But ​​I'm still a bit unclear about why the accuracy on the VizWiz dataset is so unusual, especially since I'm using the official evaluation code provided on the VizWiz website.
-Hyperparameters: I strictly adhered to the default configurations provided in your codebase. I strictly adhered to the default configurations provided in your codebase, with the only modification being the adjustment of the token-carve-image-token-numsparameter for experimental purposes.

2.3. Results on Other Benchmarks
I have also tested TokenCarve on the ​​POPE​​ benchmark for hallucination assessment. The results there appear normal and show the expected trend:
Vanilla Model: ​​84.97%​​
192 tokens: ​​85.35%​​
128 tokens: ​​85.63%​​
64 tokens: ​​84.90%​​
We have not yet tested on other benchmarks like SQA.
Thanks again for the helpful pointer—I look forward to figuring this out.

**Comment by ShawnTan86** [comment]
> Thank you again for your prompt and detailed response. We sincerely appreciate your engagement and are providing the following clarifications based on your questions.
> 
> 1. Experimental Setup Clarification
> 
> * Regarding your inquiry about our experimental setup:
> 
> ​​Model Architecture & Checkpoint:​​ We used the standard ​​LLaVA-v1.5-7B​​ model, specifically the checkpoint liuhaotian/llava-v1.5-7bdownloaded directly from the official Hugging Face repository. No modifications were made to the core architecture.
> 
> * Evaluation Metric for TextVQA and VizWiz:
>   We confirmed that the evaluation for the TextVQA and VizWiz dataset was performed using ​​exact match. And I see now that the reason for the low accuracy in my TextVQA evaluation is likely because the associated OCR data was not used. But ​​I'm still a bit unclear about why the accuracy on the VizWiz dataset is so unusual, especially since I'm using the official evaluation code provided on the VizWiz website.
>   -Hyperparameters: I strictly adhered to the default configurations provided in your codebase. I strictly adhered to the default configurations provided in your codebase, with the only modification being the adjustment of the token-carve-image-token-numsparameter for experimental purposes.
> 
> 2.3. Results on Other Benchmarks I have also tested TokenCarve on the ​​POPE​​ benchmark for hallucination assessment. The results there appear normal and show the expected trend: Vanilla Model: ​​84.97%​​192 tokens: ​​85.35%​​128 tokens: ​​85.63%​​64 tokens: ​​84.90%​​We have not yet tested on other benchmarks like SQA. Thanks again for the helpful pointer—I look forward to figuring this out.

Hi, thanks for the detailed follow-up and for sharing your clarifications.

1. **On the TextVQA results:**  
   It makes sense that the missing OCR data could have affected the accuracy. Thanks for confirming the evaluation setup.  
   Also, your **POPE** benchmark results look great — even slightly higher than those in our paper, which is encouraging to see.

2. **On the VizWiz results:**  
   In our experiments, we obtained the official scores by submitting the prediction files to **EvalAI** ([https://eval.ai/web/challenges/challenge-page/2185/my-submission](https://eval.ai/web/challenges/challenge-page/2185/my-submission)).  
   As mentioned in the *Main Results* section of the TokenCarve paper:  
   > “An interesting phenomenon is that the SQA and VizWizVQA datasets are not sensitive to token compression rate, which is present in all visual token compression methods.”  

   We also observed this insensitivity in multiple runs and across different model sizes. It seems that **VizWiz** may generally be less sensitive to visual token compression, which could explain the unusual trend you found.  
   This is indeed an interesting phenomenon — I didn’t explore it in depth, but if you make further progress or new observations in this area, I’d be very happy to discuss them.

Thanks again for your careful reproduction and valuable feedback!

**Comment by ZeroDis0bnbs** [comment]
Thank you for your previous response.​​ Could you please show me the format of the prompt after combining the TextVQA questions with the OCR tokens? Thank you, and I look forward to hearing from you.

**Comment by ShawnTan86** [comment]
> Thank you for your previous response.​​ Could you please show me the format of the prompt after combining the TextVQA questions with the OCR tokens? Thank you, and I look forward to hearing from you.

Hi, thanks for your question.

We follow the official **LLaVA** evaluation format for combining TextVQA questions with the OCR tokens. Here’s an example of the **question-file** format we use, where each data entry is on a single line:

```json
{"question_id": "003a8ae2ef43b901", "image": "003a8ae2ef43b901.jpg", "text": "What is the brand of this camera?\nReference OCR token: DAKOTA, DIGITAL, Single-Use, Camera, Pire, digitat\nAnswer the question using a single word or phrase.", "category": "default"}
{"question_id": "b9dc400eb20bad64", "image": "b9dc400eb20bad64.jpg", "text": "What does the small white text spell?\nReference OCR token: D, RUPALCON, PALCON, COPENHAGEN\nAnswer the question using a single word or phrase.", "category": "default"}
{"question_id": "2b538a43dd933fc1", "image": "2b538a43dd933fc1.jpg", "text": "What kind of beer is this?\nReference OCR token: NINK, NK, BOWING, CC, STON, SUE, ED, Sublimely, SELF, ELF-RICHEE, swAaVd, KGy, ALE\nAnswer the question using a single word or phrase.", "category": "default"}
```


You can refer to the official LLaVA documentation for more details: [LLaVA Evaluation Format](https://github.com/haotian-liu/LLaVA/blob/main/docs/Evaluation.md).
